cantor-archive/src/main/java/com/salesforce/cantor/archive/file/AbstractBaseArchiverOnFile.java

```diff
@@ -0,0 +1,88 @@
+package com.salesforce.cantor.archive.file;
+
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.ArchiveOutputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveEntry;
+import org.apache.commons.compress.archivers.tar.TarArchiveInputStream;
+import org.apache.commons.compress.archivers.tar.TarArchiveOutputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorInputStream;
+import org.apache.commons.compress.compressors.gzip.GzipCompressorOutputStream;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedOutputStream;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.salesforce.cantor.common.CommonPreconditions.checkArgument;
+import static com.salesforce.cantor.common.CommonPreconditions.checkString;
+
+public abstract class AbstractBaseArchiverOnFile {
+    protected final String baseDirectory;
+    protected final long chunkMillis;
+
+    // default to nothing for no sub directory
+    protected String subDirectory = "";
+
+    public AbstractBaseArchiverOnFile(final String baseDirectory) {
+        this(baseDirectory, 0);
+    }
+
+    public AbstractBaseArchiverOnFile(final String baseDirectory, final long chunkMillis) {
+        this.baseDirectory = baseDirectory;
+        this.chunkMillis = chunkMillis;
+    }
+
+    public Path getFile(final String fileNameFormat, final Object... args) {
```

Comment:
@AlexanderKouthoofd this can be protected, right?

Reply:
Yeah it's public because I'm using it for the S3 implementation. I could probably get around it, but that would require coping code which I wanted to keep to a minimum.